node/src/components/consensus/era_supervisor.rs

```diff
@@ -698,6 +707,10 @@ where
         era_id: EraId,
         proto_block: ProtoBlock,
     ) -> Effects<Event<I>> {
+        self.era_supervisor
+            .metrics
+            .time_of_last_proposed_block
+            .set(Timestamp::now().millis() as f64 / 1000.00);
```

Comment:
…or should we actually use the block's timestamp instead of `now()`?
Not sure: Timestamps can be faked, of course. On the other hand, the time when we get the block might be pretty different from the time where it was proposed — especially while we're catching up when joining the network.

Reply:
I use `now()` as its when the EraSupervisor 'accepts' the proto-block at that instance of time. I thought recording when we actually accept the block would be important, and we can separately note when the block was proposed.